CalendarFXView/src/main/java/com/calendarfx/view/DateControl.java

```diff
@@ -1038,6 +1037,67 @@ public String toString() {
         }
     }
 
+    /**
+     * Possible edit operations with an entry.
+     * This enum will be used as parameter of the callback set with {@link DateControl#setEntryEditPolicy}.
+     */
+    public enum EditOperation {
+
+        /**
+         * Checked if any drag/drop operation is allowed
+         */
+        DRAG_AND_DROP,
+
+        /**
+         * Checked if the start of an entry can be changed (e.g. using drag/drop).
+         * for drag and drop you also have to allow {@link #DRAG_AND_DROP}
+         */
+        CHANGE_START,
+
+        /**
+         * Checked if the end of an entry can be changed (e.g. using drag/drop)
+         * for drag and drop you also have to allow {@link #DRAG_AND_DROP}
+         */
+        CHANGE_END,
+
+        /**
+         * Checked if entry can be moved using drag/drop
+         * for drag and drop you also have to allow {@link #DRAG_AND_DROP}
+         */
+        MOVE,
+
+        /**
+         * Checked if an entry can be deleted
+         */
+        DELETE
+    }
+
+    /**
+     * If an action will be issued on an item the given instance will be asked if the action is allowed
+     */
+    private final SimpleObjectProperty<Callback<EntryEditParameter, Boolean>> entryEditPolicy = new SimpleObjectProperty<>(action -> true);
```

Comment:
On the left-hand side we always use the most generic property type. In this case "ObjectProperty" and not "SimpleObjectProperty".